libs/common/src/platform/state/storage/memory-storage.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Subject } from "rxjs";
+
+import {
+  AbstractMemoryStorageService,
+  ObservableStorageService,
+  StorageUpdate,
+} from "../../abstractions/storage.service";
+
+export class MemoryStorageService
+  extends AbstractMemoryStorageService
+  implements ObservableStorageService
+{
+  protected store: Record<string, string> = {};
+  private updatesSubject = new Subject<StorageUpdate>();
+
+  get valuesRequireDeserialization(): boolean {
+    return true;
+  }
+  get updates$() {
+    return this.updatesSubject.asObservable();
+  }
+
+  get<T>(key: string): Promise<T> {
+    if (this.store[key]) {
+      const json = this.store[key];
```

Comment:
Seems worth it to just get the value from the store a single time. 

```suggestion
    const json = this.store[key];
    if (json) {
```